sympy/matrices/matrices.py

```diff
@@ -3912,35 +3975,121 @@ def LUdecomposition_Simple(self, iszerofunc=_iszero):
+                issue=9796,
+                deprecated_since_version="1.1").warn()
+
+        if self.rows == 0 or self.cols == 0:
+            # Define LU decomposition of a matrix with no entries as a matrix
+            # of the same dimensions with all zero entries.
+            return self.zeros(self.rows, self.cols), []
+
+        lu = self.as_mutable()
+        row_swaps = []
+
+        pivot_col = 0
+        for pivot_row in range(0, lu.rows-1):
+            # Search for pivot. Prefer entry that iszeropivot determines
+            # is nonzero, over entry that iszeropivot cannot guarantee
+            # is  zero.
+            candidate_pivot_row = None
+            while pivot_col != self.cols:
+                for r in range(pivot_row, self.rows):
+                    iszeropivot = iszerofunc(lu[r, pivot_col])
+                    if iszeropivot is None and candidate_pivot_row is None:
+                        # iszerofunc could not determine if
+                        # lu[r, pivot_col] is zero.
+                        # Save this row in case we don't
+                        # find a nonzero entry during the search
+                        # of this column
+                        candidate_pivot_row = r
+                    elif iszeropivot is False:
+                        # iszerofunc determined that
+                        # lu[r, pivot_col] is zero.
+                        # Use this entry as the pivot
+                        candidate_pivot_row = r
+                        break
+                    elif iszeropivot is True:
+                        lu[r, pivot_col] = S.Zero
+                if candidate_pivot_row is None:
+                    # There is no nonzero pivot in this column.
+                    # Exited search within column because
+                    # iszeropivot() returned False for each
+                    # entry
+                    pivot_col += 1
+                else:
+                    # lu[r, pivot_col] is either non-zero,
+                    # or iszeropivot could not determine that it is
+                    # zero.
+                    break
+
+            if rankcheck and pivot_col != pivot_row:
+                # All entries including and below the pivot position are
+                # zero, which indicates that the rank of the matrix is
+                # strictly less than min(num rows, num cols)
+                # Mimic behavior of previous implementation, by throwing a
+                # ValueError.
+
+                raise ValueError("Rank of matrix is strictly less than"
+                                 " number of rows or columns."
+                                 " Pass keyword argument"
+                                 " rankcheck=False to compute"
+                                 " the LU decomposition of this matrix."
+                                 " This behavior is deprecated since sympy"
+                                 " 1.1.")
```

Comment:
I don't think the `ValueError` should include a deprecation warning aswell, especially since they've already been warned.